ggshield/scan/scannable.py

```diff
@@ -298,6 +292,52 @@ def get_filemode(line: str) -> Filemode:
 
         raise click.ClickException(f"Filemode is not detected:{line}")
 
+    @staticmethod
+    def _parse_diff_header_lines(lines: List[str]) -> Tuple[str, Filemode]:
+        """
+        Parses the part of the patch after a "diff --git " string to extract the
+        filename and filemode.
+        """
+
+        # This code is tricky because the first line of a diff looks like this:
+        #
+        #     diff --git a/path/to/filename b/path/to/filename
+        #
+        # But spaces are not escaped, so if a filename contains a space the line looks
+        # like this:
+        #
+        #     diff --git a/path/to/file name b/path/to/file name
+        #
+        # For all cases but RENAMEs, the old and the new file are the same. We take
+        # advantage of this in the _MATCH_FILENAME_RX regular expression.
+        #
+        # For RENAMEs we look at the next lines. A rename header looks like this:
+        #
+        #     diff --git a/old_name b/new_name
+        #     similarity index 90%
+        #     rename from old_name
+        #     rename to new_name
+        #
+        # We look for the "rename to " and get the new file name from it.
+        #
+        # Note: we can't use the "--- old", "+++ new" header lines because
+        # PERMISSION_CHANGE and RENAME without changes do not have these lines.
+        filemode = Commit._get_filemode(lines[1])
+        if filemode == Filemode.RENAME:
+            prefix = "rename to "
+            for line in lines[2:]:
+                if line.startswith(prefix):
+                    filename = line[len(prefix) :]
+                    break
+            else:
+                raise click.ClickException(f"Could not extract filename from {lines}")
```

Comment:
From what I understand, this should almost never happen : but we will end up logging code and possibly secrets ? Shouldn't we log the first lines only instead ?